### ISSUE TITLE
capybara rack/test browser session erasing fragment params

### ISSUE THREAD

**Issue by HoneyryderChuck** [teamcapybara/capybara#2368]
## Meta
Capybara Version:
3.32.2?

## Expected Behavior

Coming from [here](https://github.com/rack/rack/issues/1684). After some discussion with the rack maintainers, everyone's in agreement this is a capybara issue, mostly a rack test browser session one.

After submitting an HTML form, if a server responds with a 302, capybara takes care of following the redirects, as browsers do. 

However, if the server responds with a URI containing a fragment, capybara "cleans" the fragment when following the redirect. This is because [Capybara uses the rack-test methods for performing a get request](https://github.com/teamcapybara/capybara/blob/master/lib/capybara/rack_test/browser.rb#L54), which in turn end up calling `Rack::MockRequest.env_for`. This will generate a rack request env, which will not take fragments into account.

After a talk with the rack maintainers, we agree that, if capybara plays the role of browser, it should be capybara's concern to propagate the fragment, in this case.

What do you think?

**Comment by twalpole** [comment]
Please try the `rack_fragment` branch and see if that does what you expect.

**Comment by twalpole** [comment]
Please provide a self contained example showing your issue

**Comment by HoneyryderChuck** [comment]
```ruby
require "roda"
require "capybara"

class App < Roda
  route do |r|
    r.root { "This is #{request.url}" }
    r.on "fragment" do
      r.redirect "/#bang=bang"
    end
  end
end
App.plugin :common_logger 

session = Capybara::Session.new(:rack_test, App)
session.visit "/fragment"
puts session.current_url
# right now, it's http://www.example.com/
```

**Comment by twalpole** [comment]
Closed via #2369

**Comment by HoneyryderChuck** [comment]
it worked. thx @twalpole !
